### ISSUE TITLE
Support for unknown fields (text input) in left operand.

### ISSUE THREAD

**Issue by sivaram000** [react-querybuilder/react-querybuilder#693]
**Description of the feature**

The left operand of the rule is always dropdown for which we have to define fields beforehand. Is it possible to add input field so the user will add any text in key then select condition from list and then add value. 

<img width="480" alt="Screenshot 2024-05-06 at 12 18 51 PM" src="https://github.com/react-querybuilder/react-querybuilder/assets/89124860/5d5147fc-46c9-4078-9099-0500726688b0">

Please point me to the documentation incase its already supported.

**Comment by jakeboone02** [comment]
This is a relatively simple thing to do (see below), but when I tried to come up with a working sandbox I found out it isn't actually possible in the current version due to a bug. I'll fix the bug and probably release a patch later today after merging #692.

Once the bug is fixed, you'll be able to set the `field` to anything you want from a custom field selector. The example below renders a free-form text box instead of a select box, so whatever the user types is what gets stored in the rule's `field` property. Another option would be to use something like `react-select` that handles creation of new options on the fly.

```tsx
const FieldSelector = (props: FieldSelectorProps) => (
  <input
    type="text"
    value={props.value}
    onChange={e => props.handleOnChange(e.target.value)}
  />
);
```
